h2gis-utilities/src/main/java/org/h2gis/utilities/JDBCUtilities.java

```diff
@@ -868,6 +868,293 @@ public static String createTableDDL(ResultSet resultSet, String outputTableName)
+            ps.setObject(2, table.getSchema("public"));
+            ps.setObject(3, table.getTable());
+            ResultSet rs = ps.executeQuery();
+            return rs.next();
+        }
+    }
+
+    /**
+     * Create an index on the given column of the given table on the given connection.
+     * @param connection Connection to access to the desired table.
+     * @param table      Table containing the column to index.
+     * @param columnName Name of the column to index.
+     * @return True if the column have been indexed, false otherwise.
+     * @throws SQLException Exception thrown on SQL execution error.
+     */
+    public static boolean createIndex(Connection connection, TableLocation table, String columnName) throws SQLException {
+        if(connection == null || table == null || columnName == null){
+            throw new SQLException("Unable to create an index");
+        }
+        boolean isH2 = isH2DataBase(connection);
+        connection.createStatement().execute("CREATE INDEX IF NOT EXISTS " + table.toString(isH2) + "_" + columnName +
+                " ON " + table.toString(isH2) + " USING BTREE (" + TableLocation.capsIdentifier(columnName, isH2) + ")");
+        return true;
+    }
+
+    /**
+     * Create an index on the given column of the given table on the given connection.
+     * @param connection Connection to access to the desired table.
+     * @param table      Name of the table containing the column to index.
+     * @param columnName Name of the column to index.
+     * @return True if the column have been indexed, false otherwise.
+     * @throws SQLException Exception thrown on SQL execution error.
+     */
+    public static boolean createIndex(Connection connection, String table, String columnName) throws SQLException {
+        return createIndex(connection, TableLocation.parse(table, isH2DataBase(connection)), columnName);
+    }
+
+    /**
+     * Create a spatial index on the given column of the given table on the given connection.
+     * @param connection Connection to access to the desired table.
+     * @param table      Table containing the column to index.
+     * @param columnName Name of the column to index.
+     * @return True if the column have been indexed, false otherwise.
+     * @throws SQLException Exception thrown on SQL execution error.
+     */
+    public static boolean createSpatialIndex(Connection connection, TableLocation table, String columnName) throws SQLException {
+        if(connection == null || table == null || columnName == null){
+            throw new SQLException("Unable to create a spatial index");
+        }
+        boolean isH2 = isH2DataBase(connection);
+        if (isH2) {
+            connection.createStatement().execute("CREATE INDEX IF NOT EXISTS " + table.toString(isH2) + "_" + columnName +
+                    " ON " + table.toString(isH2) + " USING RTREE (" + TableLocation.capsIdentifier(columnName, isH2)  + ")");
```

Comment:
You should also use documented and supported `CREATE SPATIAL INDEX` instead of unsupported `USING RTREE` here.

Reply:
Ok, I will use `CREATE SPATIAL INDEX` syntax, thank you.